components/auth/RegisterAgreement.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import palette from '../../lib/styles/palette';
+import { CheckboxItem } from '.';
+import { registerFormItems } from '../../lib/texts/texts';
+import { RoundButton } from '..';
+
+type AgreementProps = {
+  toPageNext: Function;
+  registerForm: object;
+  setRegisterForm: Function;
+};
+
+export const RegisterAgreement = ({
+  toPageNext,
+  registerForm,
+  setRegisterForm,
+}: AgreementProps) => {
+  const [isShowAll, setIsShowAll] = useState(false);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [isCheckedAll, setIsCheckedAll] = useState(false);
+  // BasicTerm, PersonalInfo, SMS, Email
+  const [checkedList, setCheckedLists] = useState([false, false, false, false]);
+
+  const showAll = () => {
+    setIsShowAll(true);
+  };
+
+  const handleCheckAll = () => {
+    setCheckedLists([
+      !isCheckedAll,
+      !isCheckedAll,
+      !isCheckedAll,
+      !isCheckedAll,
+    ]);
```

Comment:
이렇게 바꿔보면 어떨까요? 더 깔끔하고, State의 길이가 늘어나는 경우에 유연하게 대처할 수 있을것 같습니다.
```suggestion
    setCheckedLists((prev) => prev.map(!isCheckedAll));
```